src/Component/InlineEdit.php

```diff
@@ -0,0 +1,179 @@
+     *
+     * @var null|string The name of the field.
+     */
+    public $modelField = null;
+
+    /**
+     * Whether component should save it's value when input get blur.
+     *
+     * @var bool
+     */
+    public $saveOnBlur = true;
+
+    /**
+     * Default css for the input div.
+     *
+     * @var string
+     */
+    public $inputCss = 'ui right icon input';
+
+    /**
+     * Initialization.
+     */
+    public function init()
+    {
+        parent::init();
+        $this->cb = $this->add('jsCallback');
+    }
+
+    /**
+     * Set Model of this View.
+     *
+     * @param \atk4\data\Model $model
+     *
+     * @return \atk4\data\Model
+     */
+    public function setModel(\atk4\data\Model $model)
+    {
+        parent::setModel($model);
+        $this->modelField = $this->modelField ? $this->modelField : $this->model->title_field;
+        if ($this->autoSave && $this->model->loaded()) {
+            if ($this->cb->triggered()) {
+                $value = $_POST['value'] ? $_POST['value'] : null;
+                $this->cb->set(function () use ($value) {
+                    try {
+                        $this->model[$this->modelField] = $value;
+                        $this->model->save();
+
+                        return $this->jsSuccess('Update successfully');
+                    } catch (\atk4\data\Exception $e) {
+                        return $this->jsError($e->getMessage());
+                    } catch (\Error $e) {
+                        return $this->jsError($e->getMessage());
+                    }
+                });
+            }
+        }
+
+        return $this->model;
+    }
+
+    /**
+     * onChange handler.
+     * You may supply your own function to handle update.
+     * The function will receive two params:
+     *  id: The record id;
+     *  value:  the new input value.
+     *
+     * @param callable $fx
+     */
+    public function onChange($fx)
+    {
+        if (is_callable($fx) && !$this->autoSave) {
+            if ($this->cb->triggered()) {
+                $id = $_POST['id'] ? $_POST['id'] : null;
+                $value = $_POST['value'] ? $_POST['value'] : null;
+                $this->cb->set(function () use ($fx, $id, $value) {
+                    return call_user_func($fx, $id, $value);
+                });
+            }
+        }
+    }
+
+    /**
+     * On success message.
+     *
+     * @param string $message
+     *
+     * @return \atk4\ui\jsToast
+     */
+    public function jsSuccess($message)
+    {
+        return new \atk4\ui\jsToast([
+                                        'title'   => 'Success',
+                                        'message' => $message,
+                                        'class'   => 'success',
+                                    ]);
+    }
+
+    /**
+     * On error message.
+     *
+     * @param string $message
+     *
+     * @return \atk4\ui\jsToast
+     */
+    public function jsError($message)
```

Comment:
is that consistent with $form->error() ?

Reply:
Update to use field validation in model.